src/index.ts

```diff
@@ -27,8 +27,8 @@ class LAN {
   private readLegacy = (buf: any) => {
     if (buf.loopback) return;
     const address = buf.toString();
-    const peer = Ref.parseAddress(address);
-    if (peer?.key !== this.ssb.id) {
+    const peerKey = Ref.getKeyFromAddress(address);
+    if (peerKey !== this.ssb.id) {
```

Comment:
could we do `peerKey && peerKey !== this.ssb.id`? Because `peer?.key` actually contained a check whether the parsing succeeded or not. But now, if `peerKey` is undefined, this if condition would pass.

Reply:
Sweet, thanks for the fix!